Dev callbackimplementation

7178301 opened the pull request.

7178301 committed:
NEW: Sensor Callback Feature

7178301 committed:
NEW: SDKAdaptor Callbacks & SDKAdaptor Flight Commands Stack

7178301 committed:
ENHANCE:  Expanded Logging & LogCallback Functionality

7178301 committed:
ENHANCE: Android Logging

cameroncros commented:
Rather than pushing a position and a delay, push a delay seperately.
Also should have a way to push a camera event or other sensor capture event
empty while loop in TestFlight1 should have a Thread.sleep(10) to avoid thrashing the cpu
for cahngeAltitudeDisplacement etc, perhaps the callback should be optional and the second arguement. ie: changeAltitudeDisplacement(double alt, SDKAdaptorCallback callback = null); this allows the end user to ignore the callback if they dont care about the result.
With adding the callback, the return values dont make much sense (could return true and fail in the callback, perhaps the return values should be removed
Im not sure max speed and max rotate speed should be in SDKAdaptor, the most likely arent universal constants, so should be on a per adaptor basis
the sdkAdaptor functions, you should just be passing the callback down the the next function, not creating a new callback to wrap the old callback
 this:
final public boolean changeAltitudeDisplacement(final SDKAdaptorCallback sdkAdaptorCallback, double altitude) {
        return flyTo(new SDKAdaptorCallback() {
            @Override
            public void onResult(boolean booleanResult, String stringResult) {
                sdkAdaptorCallback.onResult(booleanResult, stringResult);
            }
        }, new PositionDisplacement(0, 0, altitude, new Angle(0), new Angle(0), new Angle(0)));
    }
becomes:
final public boolean changeAltitudeDisplacement(final SDKAdaptorCallback sdkAdaptorCallback, double altitude) {
        return flyTo(sdkAdaptorCallback
        , new PositionDisplacement(0, 0, altitude, new Angle(0), new Angle(0), new Angle(0)));
    }

Should make sure that null callbacks are handled correctly as well.
Why was goHome changed to final? Do we not want the adaptors to override it?

7178301 committed:
ENHANCE: Removed Unnecessary Extra SDKAdaptorCallback Instance

7178301 committed:
ENHANCE: Position null Checks & TestFlight Thread Handling

7178301 commented:
Java does not support parameter default values, instead uses overloading.
Callback & return values should be set in adapters and be set to the same value
MaxSpeed and MaxRotate should be individual values to each drone, yet exist universally for all drones
GoHome currently returns the drone to the set home variable via FlyTo which is overridden, therefor shouldn't final is adequate. Only problem would be if the drone can't follow the LOS home, therefor whole functions should be abstract with no implementation.

SDKAdaptorStack currently being expanded on to allow more SDKAdaptor commands to be stacked.
Also looking into a potential sensor event stack.

7178301 committed:
ENHANCE: SDKAdaptorFlightStack Threading & Delay Push

7178301 committed:
ENHANCE: SDKAdaptorCallback null checks & Simulator Logging

7178301 committed:
NEW: Bearing Abstract & Android Bearing Sensor

7178301 committed:
ENHANCE: Changed Network Servers To Single Thread (Security & Reliability)

7178301 committed:
ENHANCE: Added logging throughout

7178301 committed:
NEW: Sensor Events for SDKAdaptorStack

7178301 committed:
ENHANCE: Push Whole Task to SKDAdaptorTaskStack

7178301 committed:
ENHANCE:  Network Security

Communication Protocols don't need to know about everything, just how to
execute commands.

7178301 committed:
ENHANCE: Stack now Thread Tollerable

7178301 committed:
QUALITY: Code Formatting Standardized

cameroncros commented:
if(sdkAdaptorCallback==null)
            throw new IllegalArgumentException("Callback Required");
^^ not really what i meant by dealing with null callbacks, i meant if the callback is null then don't bother calling the callback, There is likely a case where you don't care what the callback result is (extreme confidence perhaps :P ) 

Otherwise i am happy to merge this when you are

7178301 committed:
ENHANCE: Allow Null Callbacks & new function methods in SDKAdaptorStack
